### PR TITLE
odafileconverter: init at 21.7.0.0

### PR DESCRIPTION
**Pull request by nagisa**
###### Motivation for this change

This is _the_ way to convert AutoCAD DWG files to something more open on Linux. It can work both as a stand-alone tool as well as used transparently by freecad and similar tooling to enable DWG support in them.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/odaconverter/default.nix
Comment (ymarkus) [inline pkgs/applications/graphics/odaconverter/default.nix:42]: Please add a maintainer as per [the nixpkgs manual](https://nixos.org/nixpkgs/manual/#reviewing-contributions-new-packages):
> A maintainer must be set. This can be the package submitter or a community member that accepts to take maintainership of the package.

If you aren't yet a maintainer, you would have to add yourself to `maintainers/maintainer-list.nix`

As an example, this is how I do it: 
```
meta = with stdenv.lib; {
  [...]
  maintainers = with maintainers; [ ymarkus ];
}
```

File: pkgs/applications/graphics/odaconverter/default.nix
Comment (ymarkus) [inline pkgs/applications/graphics/odaconverter/default.nix:45]: As you have included `stdenv.lib` above, change this to:
```
license = licenses.unfree;
```

File: pkgs/applications/graphics/odaconverter/default.nix
Comment (ymarkus) [inline pkgs/applications/graphics/odaconverter/default.nix:5]: Please add indentation here.

**Review by ymarkus — CHANGES_REQUESTED** [review]
##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] phases are respected

##### Possible improvements
Please see my comments

##### Comments
I'm not sure if the package name `odafileconverter` might be more accurate, but I think this name is fine.

File: pkgs/applications/graphics/odaconverter/default.nix
Comment (ymarkus) [inline pkgs/applications/graphics/odaconverter/default.nix:9]: Please add a `name` attribute, such as:
```
name = "${pname}-${version}";
```

**Review by ymarkus — CHANGES_REQUESTED** [review]
Forgot the name attribute in my last review. Please see my comment.

File: pkgs/applications/graphics/odaconverter/default.nix
Comment (nagisa) [reply pkgs/applications/graphics/odaconverter/default.nix:9]: Is it necessary? I thought `pname` + `version` was all you needed?

File: pkgs/applications/graphics/odaconverter/default.nix
Comment (nagisa) [reply pkgs/applications/graphics/odaconverter/default.nix]: Done.

File: pkgs/applications/graphics/odaconverter/default.nix
Comment (nagisa) [reply pkgs/applications/graphics/odaconverter/default.nix]: Done.

File: pkgs/applications/graphics/odaconverter/default.nix
Comment (nagisa) [reply pkgs/applications/graphics/odaconverter/default.nix]: Done.

File: pkgs/applications/graphics/odaconverter/default.nix
Comment (ymarkus) [reply pkgs/applications/graphics/odaconverter/default.nix]: Seems like you're right, sorry. [This section](https://nixos.org/nixpkgs/manual/#sec-package-naming) of the nixpkg manual made it seem like it is required, but it isn't.

**Review by ymarkus — APPROVED** [review]
All good now from my side. Built & tested on `x86_64-linux`.

##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] phases are respected

**Review by Ekleog — APPROVED** [review]
LGTM! I think this is ready to merge once the comment about where the version number comes from is added, though it'd be best if you could also have a look at the other comments

File: pkgs/applications/graphics/odafileconverter/default.nix
Comment (nagisa) [reply pkgs/applications/graphics/odafileconverter/default.nix:7]: They are pretty bad at software distribution honestly speaking. Not only do they not publish the version number (gotta dig for it in the deb metadata) but they also do not maintain stable URI for the past releases.

File: pkgs/applications/graphics/odafileconverter/default.nix
Comment (nagisa) [reply pkgs/applications/graphics/odafileconverter/default.nix]: Looks like either way works. Adjusted.

File: pkgs/applications/graphics/odafileconverter/default.nix
Comment (nagisa) [reply pkgs/applications/graphics/odafileconverter/default.nix]: Done.

**Comment by Ekleog** [comment]
Cool, thank you!

**Pull request merged**
